app/invites/server/functions/useInviteToken.js

```diff
@@ -16,18 +16,24 @@ export const useInviteToken = (userId, token) => {
 	const { inviteData, room } = validateInviteToken(token);
 
 	const user = Users.findOneById(userId);
+	Users.updateInviteToken(user._id, token);
```

Comment:
Rather than save the token id in the user, could add a list of room's ids in the user, grab that info [here](https://github.com/RocketChat/Rocket.Chat/blob/3a5e86400d73896fe4dd5e1eb12101b6448cf17a/app/lib/server/functions/addUserToDefaultChannels.js#L6) merging that rooms with the default ones.

Reply:
what about if the list of default channels change in the meantime of user register and set his username? I think it is safer to keep just the invite token and It is also less information to be saved. this also helps identifying how the user was registered.

Reply:
I was not meaning to save all the channels, but extra channels to be added, in this case always one, but be a list to be prepared.

Reply:
we'll evaluate this again in the future